packages/grid/x-data-grid/src/DataGrid/useDataGridProps.ts

```diff
@@ -76,10 +75,6 @@ export const DATA_GRID_PROPS_DEFAULT_VALUES: DataGridPropsWithDefaultValues = {
 };
 
 export const useDataGridProps = <R extends GridValidRowModel>(inProps: DataGridProps<R>) => {
-  if (inProps.pageSize! > MAX_PAGE_SIZE) {
```

Comment:
Why are we removing this check?

Reply:
This check was enough for the Community plan since the only way to change the page size was to pass a prop.
And since the prop calls the API method under the hood, I've moved the check to the method, so both the method and the prop won't work if the check fails.
https://github.com/cherniavskii/material-ui-x/blob/apiRef-for-community/packages/grid/x-data-grid/src/hooks/features/pagination/useGridPageSize.ts#L58-L65